File: android/app/src/main/java/com/familyfarmrn/FirebaseEvent.kt
package com.familyfarmrn

import com.facebook.react.bridge.WritableMap

class FirebaseEvent(private val eventName: String,
                    private val eventBody: WritableMap,
                    private val firebaseAppName: String? = null): NativeEvent {


    override fun getEventName(): String {
        return this.eventName
    }

    override fun getEventBody(): WritableMap {
        return this.eventBody
    }

    override fun getFirebaseAppName(): String? {
        return this.firebaseAppName
    }
}

File: ios/FirebaseEvent.swift
protocol NativeEvent {
    func getEventName() -> String
    func getEventBody() -> [String: Any]
}

class FirebaseEvent: NativeEvent {
    private let eventName: String
    private let eventBody: [String: Any]

    init(eventName: String, eventBody: [String: Any]) {
        self.eventName = eventName
        self.eventBody = eventBody
    }

    func getEventName() -> String {
        return self.eventName
    }

    func getEventBody() -> [String: Any] {
        return self.eventBody
    }
}
